### PR TITLE
feat: Add TextChoices validation to Ticket priority field

### PR DESCRIPTION
**Pull request by swapnil-turing**
- Add Priority TextChoices enum: low, medium, high, urgent
- Update factory to use FuzzyChoice for priority
- Add test for invalid priority rejection
- Add test for default priority behavior
- Update all tests to use enum values for priority

# Fixes Issue

**My PR closes #issue_number_here**

# 👨‍💻 Changes proposed(What did you do ?)

# ✔️ Check List (Check all the applicable boxes)

- [] My code follows the code style of this project.
- [] This PR does not contain plagiarized content.
- [] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/ticket/tests.py [hunk 1]
```diff
@@ -34,117 +34,105 @@ def setUp(self):
...
             "status": "Updated Status",
-            "priority": "Updated Priority",
+            "priority": Ticket.Priority.URGENT,
             "assignee": self.user2.id,
         }
         response = self.authenticated_client.put(url, payload)
-
-        # THEN we should get a 200 response
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.data["title"], "Updated Title")
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:104]: value:good-to-have; category:bug; feedback:Same reasoning as comment 3 unrelated field assertions should be preserved to maintain coverage.

File: augment-store/server/ticket/tests.py [hunk 2]
```diff
@@ -34,117 +34,105 @@ def setUp(self):
...
     def test_ticket_detail(self):
-        # GIVEN an authenticated user exists
-        # WHEN we make a get request to retrieve ticket details
         url = reverse("v1:ticket:ticket_detail", args=[self.ticket.id])
         response = self.authenticated_client.get(url)
-
-        # THEN we should get a 200 response
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.data["id"], str(self.ticket.id))  
         self.assertEqual(response.data["title"], "Test Title")
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:86]: value:good-to-have; category:bug; feedback:Removing unrelated field assertions reduces test coverage. While unlikely to cause a production issue, it weakens regression detection.

File: augment-store/server/ticket/tests.py [hunk 3]
```diff
@@ -34,117 +34,105 @@ def setUp(self):
...
+
+    def test_create_ticket_default_priority(self):
+        url = reverse("v1:ticket:create_ticket")
+        payload = {
+            "title": "Default Priority Ticket",
+            "description": "Should get default priority",
+            "status": "open",
+            "assignee": str(self.user.id),
+        }
+        response = self.authenticated_client.post(url, payload)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:71]: value:useful; category:bug; feedback:The weak assertion would have passed even if the API returned an unexpected default. This catch is especially important since the default is LOW (a deliberate design choice that should be tested explicitly).

File: augment-store/server/ticket/migrations/0002_alter_ticket_priority.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='ticket',
+            name='priority',
+            field=models.CharField(
+                choices=[('low', 'Low'), ('medium', 'Medium'), ('high', 'High'), ('urgent', 'Urgent')],
+                default='low',
+                max_length=20,
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/migrations/0002_alter_ticket_priority.py:27]: value:delightful; category:bug; feedback:Same migration failure risk as the status field  existing rows with long priority values would cause the ALTER to fail on strict databases.

File: augment-store/server/ticket/migrations/0002_alter_ticket_priority.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from django.db import migrations, models
+
+VALID_PRIORITIES = ['low', 'medium', 'high', 'urgent']
+
+def normalize_priority(apps, schema_editor):
+    Ticket = apps.get_model('ticket', 'Ticket')
+    Ticket.objects.exclude(priority__in=VALID_PRIORITIES).update(priority='low')
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/migrations/0002_alter_ticket_priority.py]: value:delightful; category:bug; feedback:NULL values from manual DB inserts would slip through the exclude() filter and cause the AlterField to fail on strict databases. This edge case is easy to miss since Django models normally prevent NULLs, but direct DB access bypasses those guards.

File: augment-store/server/ticket/tests.py [hunk 6]
```diff
@@ -34,117 +34,107 @@ def setUp(self):
...
         payload = {
             "title": "New Ticket",
             "description": "New Ticket Description",
             "status": "New Ticket Status",
-            "priority": "New Ticket Priority",
+            "priority": Ticket.Priority.MEDIUM,
             "assignee": str(self.user.id),
         }
         response = self.authenticated_client.post(url, payload) 
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:47]: value:useful; category:bug; feedback:Without this assertion, the test would pass even if the API silently applied the default LOW priority instead of the submitted MEDIUM value. This is especially important given that the default is already LOW, not MEDIUM.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/ticket/tests.py [hunk 7]
```diff
@@ -34,117 +34,108 @@ def setUp(self):
...
+        url = reverse("v1:ticket:create_ticket")
+        payload = {
+            "title": "Bad Ticket",
+            "description": "Bad Description",
+            "status": "open",
+            "priority": "not_a_valid_priority",
+            "assignee": str(self.user.id),
+        }
+        response = self.authenticated_client.post(url, payload)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:60]: value:useful; category:bug; feedback:Same pattern as the status branc without asserting the specific field, the test could falsely pass if a different validation error (e.g., missing title) caused the 400 response.

File: augment-store/server/ticket/tests.py [hunk 8]
```diff
@@ -2,9 +2,9 @@
 from django.urls import reverse
 from rest_framework import status
 from .factory import TicketFactory, CommentFactory
+from .models import Ticket
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:5]: value:good-to-have; category:documentation; feedback:Valid housekeeping point, issue auto-closing won't work with placeholder text. No runtime impact but hurts traceability.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/ticket/tests.py [hunk 9]
```diff
@@ -23,7 +23,7 @@ def setUp(self):
             title="Test Title",
             description="Test Description",
             status="Test Status",
-            priority="Test Priority",
+            priority=Ticket.Priority.HIGH,
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:27]: value:useful; category:bug; feedback:Same issue as the status branch without re-authenticating, API calls run as a different user than the one in payloads/assertions, reducing test signal and potentially masking authorization bugs.

File: augment-store/server/ticket/migrations/0002_alter_ticket_priority.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
+from django.db import migrations, models
+from django.db.models import Q
+
+VALID_PRIORITIES = ['low', 'medium', 'high', 'urgent']
+
+def normalize_priority(apps, schema_editor):
+    Ticket = apps.get_model('ticket', 'Ticket')
+    Ticket.objects.filter(
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/migrations/0002_alter_ticket_priority.py]: value:good-to-have; category:bug; feedback:Same best practice as the status branch — ensures data migration targets the correct database in multi-DB setups. No impact in single-DB projects but follows Django's documented pattern.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
